packages/aragon-wrapper/src/identity/LocalLabelAddressProvider.js

```diff
@@ -0,0 +1,67 @@
+import { BehaviorSubject } from 'rxjs/Rx'
+import Cache from '../cache'
+import AddressIdentityProvider from './AddressIdentityProvider'
+
+/**
+ * An local label identity provider for addresses
+ *
+ * @class LocalLabelIdentityProvider
+ */
+export default class LocalLabelIdentityProvider extends AddressIdentityProvider {
+  /**
+   * Create a new local label identity provider attached to a globally-stored cache.
+   *
+   * @param {Object} [target=window.parent] An window implementing the postMessage API.
+   */
+  constructor () {
+    super()
+    this.labelCache = new Cache('localAddressLabel')
+  }
+
+  async init () {
+    await this.labelCache.init()
+
+    this.labels = new BehaviorSubject(await this.labelCache.getAll())
+      .scan((labels, modifier) => modifier(labels))
```

Comment:
The modifier is a bit of an "escape hatch", so that we can give any function we want into `scan()`. In the case of `modify()`, we provide a function that closes over both the local `address` and `metadata`, which wouldn't be available here in `init()`.

Technically we could just pass the `address` and `metadata` as parameters to `scan()`, instead of the function, and hardcode the `scan()` such that it's only allowed the `labels[address] = metadata` transform. No real preference over either.